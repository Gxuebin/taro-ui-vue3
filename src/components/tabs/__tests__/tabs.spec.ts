import { mount } from '@vue/test-utils'
import AtTabs from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtTabs as any, {
    slots,
    props: { ...values },
  })
}

const tabList = [{ title: '标签页1' }, { title: '标签页2' }, { title: '标签页3' }]
describe('AtTabs', () => {
  it('should render default AtTabs', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- tabList', () => {
    const wrapper = factory({ tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- current', () => {
    const wrapper = factory({ tabList: tabList, current: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- swipeable', () => {
    const wrapper = factory({ tabList: tabList, swipeable: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  // it('should render prop -- scroll', () => {
  //   const wrapper = factory({ tabList: tabList, scroll: true })
  //   expect(wrapper.element).toMatchSnapshot()
  // })

  it('should render prop -- tabDirection', () => {
    const wrapper = factory({ tabList: tabList, tabDirection: 'vertical' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- animated', () => {
    const wrapper = factory({ tabList: tabList, animated: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- height', () => {
    const wrapper = factory({ tabList: tabList, tabDirection: 'vertical', height: '300px' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtTabs Behavior', () => {
  it('should trigger onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, tabList: tabList })
    wrapper.find('.at-tabs .at-tabs__item').trigger('tap')
    expect(onClick).toBeCalled()
  })
})
