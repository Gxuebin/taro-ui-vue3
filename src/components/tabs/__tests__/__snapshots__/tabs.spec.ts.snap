// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabs should render default AtTabs 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  />
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- animated 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px); transition: unset;"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- class 1`] = `
<view
  class="test at-tabs at-tabs--horizontal at-tabs--WEB test"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  />
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- current 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="transform: translate3d(-200%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- height 1`] = `
<view
  class="at-tabs at-tabs--vertical at-tabs--WEB"
  style="height: 300px;"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="height: 300px; transform: translate3d(0px, -0%, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="height: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- style 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
  style="color: red;"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  />
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 0%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- swipeable 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- tabDirection 1`] = `
<view
  class="at-tabs at-tabs--vertical at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="transform: translate3d(0px, -0%, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="height: 300%;"
    />
    
    
  </view>
</view>
`;

exports[`AtTabs should render prop -- tabList 1`] = `
<view
  class="at-tabs at-tabs--horizontal at-tabs--WEB"
>
  <view
    class="at-tabs__header"
    id="tabs-AOTU2018"
  >
    <view
      class="at-tabs__item at-tabs__item--active"
      id="tabtabs-AOTU20180"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页1
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20181"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页2
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
    <view
      class="at-tabs__item"
      id="tabtabs-AOTU20182"
    >
      <text
        style="white-space: nowrap;"
      >
        标签页3
      </text>
      <view
        class="at-tabs__item-underline"
      />
    </view>
  </view>
  <view
    class="at-tabs__body"
    style="transform: translate3d(-0%, 0px, 0px);"
  >
    <view
      class="at-tabs__underline"
      style="width: 300%;"
    />
    
    
  </view>
</view>
`;
